src/game/game.cpp

```diff
@@ -811,8 +811,8 @@ Thing* Game::internalGetThing(Player* player, const Position& pos, int32_t index
 		}
 
 		int32_t subType;
```

Comment:
this variable is never referenced with your new renaming.
Plus I'd like you to test the following cases:

1. Npc that sells specific fluid type (such as mug of water or wine)
2. Npc that buys specific fluid types (such as golden mug of water)
3. Npc that buys golden mug (any subtype should be sellable, with or without liquid)